consensus/tests/common.rs

```diff
@@ -480,7 +727,79 @@ mod tests {
         let result_state = select_result.0.get(0).unwrap();
         assert_eq!(
             result_state.body.consensus_state.blockchain_length,
-            Length(77748)
+            Length(113267)
+        );
+        assert_eq!(result_state, chain_a.0.get(0).unwrap());
+    }
+
+    #[test]
+    #[wasm_bindgen_test]
+    //Current chain and candidate chain are long range fork
+    //Chain A: https://storage.googleapis.com/mina_network_block_data/mainnet-113267-3NLenrog9wkiJMoA774T9VraqSUGhCuhbDLj3JKbEzomNdjr78G8.json
+    //Chain B: https://storage.googleapis.com/mina_network_block_data/mainnet-77748-3NKaBJsN1SehD6iJwRwJSFmVzJg5DXSUQVgnMxtH4eer4aF5BrDK.json
+    //Current chain: Chain A
+    //Candidate chains: Chain B (Having very less number of sub window densities)
+    fn adversary_test_select_secure_chain_with_less_number_of_sub_window_densities() {
+        let mut chain_a = ProtocolStateChain::default();
+        let mut consensus_state = ConsensusState::default();
+        consensus_state.epoch_count = Length(23);
+        consensus_state.min_window_density = Length(14);
+        consensus_state.sub_window_densities = vec![
+            Length(7),
+            Length(2),
+            Length(2),
+            Length(5),
+            Length(6),
+            Length(7),
+            Length(5),
+            Length(7),
+            Length(5),
+            Length(5),
+            Length(5),
+        ];
+        consensus_state.curr_global_slot = GlobalSlot {
+            slot_number: GlobalSlotNumber(167176),
+            slots_per_epoch: Length(7140),
+        };
+        consensus_state.blockchain_length = Length(113267);
+        let mut prot_state = ProtocolState::default();
+        prot_state.body.consensus_state = consensus_state;
+        chain_a.push(prot_state).unwrap();
+
+        let mut chain_b = ProtocolStateChain::default();
+        let mut consensus_state = ConsensusState::default();
+        consensus_state.epoch_count = Length(15);
+        consensus_state.min_window_density = Length(33);
+        consensus_state.sub_window_densities = vec![
+            Length(6),
+            Length(1),
+            Length(3),
+            // Below sub window densities are missing, resulting in index out of bound error
```

Comment:
Are these missing from the block json repr or it's a delibrate removal to represent adversarial test case?

Reply:
delibrate removal to represent adversarial test case